src/ceph-daemon

```diff
@@ -0,0 +1,924 @@
+    subprocess.check_output(['systemctl', 'start', unit_name])
+
+def install_base_units(fsid):
+    """
+    Set up ceph.target and ceph-$fsid.target units.
+    """
+    existed = os.path.exists(args.unit_dir + '/ceph.target')
+    with open(args.unit_dir + '/ceph.target.new', 'w') as f:
+        f.write('[Unit]\n'
+                'Description=all ceph service\n'
+                '[Install]\n'
+                'WantedBy=multi-user.target\n')
+        os.rename(args.unit_dir + '/ceph.target.new',
+                  args.unit_dir + '/ceph.target')
+    if not existed:
+        subprocess.check_output(['systemctl', 'enable', 'ceph.target'])
+        subprocess.check_output(['systemctl', 'start', 'ceph.target'])
+
+    existed = os.path.exists(args.unit_dir + '/ceph-%s.target' % fsid)
+    with open(args.unit_dir + '/ceph-%s.target.new' % fsid, 'w') as f:
+        f.write('[Unit]\n'
+                'Description=ceph cluster {fsid}\n'
+                'PartOf=ceph.target\n'
+                'Before=ceph.target\n'
+                '[Install]\n'
+                'WantedBy=multi-user.target ceph.target\n'.format(
+                    fsid=fsid)
+        )
+        os.rename(args.unit_dir + '/ceph-%s.target.new' % fsid,
+                  args.unit_dir + '/ceph-%s.target' % fsid)
+    if not existed:
+        subprocess.check_output(['systemctl', 'enable', 'ceph-%s.target' % fsid])
+        subprocess.check_output(['systemctl', 'start', 'ceph-%s.target' % fsid])
+
+def get_unit_file(fsid):
+    u = """[Unit]
+Description=Ceph daemon for {fsid}
+
+# According to:
+#   http://www.freedesktop.org/wiki/Software/systemd/NetworkTarget
+# these can be removed once ceph-mon will dynamically change network
+# configuration.
+After=network-online.target local-fs.target time-sync.target
+Wants=network-online.target local-fs.target time-sync.target
+
+PartOf=ceph-{fsid}.target
+Before=ceph-{fsid}.target
+
+[Service]
+LimitNOFILE=1048576
+LimitNPROC=1048576
+EnvironmentFile=-/etc/environment
+ExecStartPre=-/usr/bin/podman rm ceph-{fsid}-%i
+ExecStartPre=-mkdir -p /var/run/ceph
+ExecStart={data_dir}/{fsid}/%i/cmd
+ExecStop=-/usr/bin/podman stop ceph-{fsid}-%i
+ExecStopPost=-/bin/rm -f /var/run/ceph/{fsid}-%i.asok
+Restart=on-failure
+RestartSec=10s
+TimeoutStartSec=120
+TimeoutStopSec=15
+StartLimitInterval=30min
+StartLimitBurst=5
+
+[Install]
+WantedBy=ceph-{fsid}.target
+""".format(fsid=fsid, data_dir=args.data_dir)
+    return u
+
+def gen_ssh_key(fsid):
+    tmp_dir = tempfile.TemporaryDirectory()
+    path = tmp_dir.name + '/key'
+    subprocess.check_output([
+        'ssh-keygen',
+        '-C', 'ceph-%s' % fsid,
+        '-N', '',
+        '-f', path
+    ])
+    with open(path, 'r') as f:
+        secret = f.read()
+    with open(path + '.pub', 'r') as f:
+        pub = f.read()
+    os.unlink(path)
+    os.unlink(path + '.pub')
+    tmp_dir.cleanup()
+    return (secret, pub)
+
+##################################
+
+class CephContainer:
```

Comment:
remoto has the ability to talk to containers as well with the same API, so you could remove this and connect to the container

Reply:
Hm, that would make us dependent on remoto on all hosts, not just on the mgr host, as this script is also supposed to be executable on any given host

Reply:
I guess the question would be why this script needs to run remotely at all?